aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLRequestFactory.java

```diff
@@ -468,4 +458,17 @@ private static String getModelFields(Class<? extends Model> clazz, int levelsDee
 
         return result.toString();
     }
+
+    static final class InvalidGraphQLRequestRuntimeException extends RuntimeException {
```

Comment:
I changed this to use `IllegalStateException`. I talked to @TrekSoft about this offline and the idea here is that those methods should only throw errors in case there's a problem with the ode during development time. Only a faulty `ModelSchema` will result in that code throwing an error which means either the CLI generated something wrong, that although it compiles, our code is not prepare to handle (a bug) or customers changed their models manually and produced something invalid (highly unlikely, can't think of a real scenario here).

Therefore, I think it's fair to have a `RuntimeException` here, no app user input can really result in a failure, at least not in this layer of the logic.